graphqlapi/src/main/java/io/stargate/graphql/schema/FieldFilterInputTypeCache.java

```diff
@@ -0,0 +1,140 @@
+package io.stargate.graphql.schema;
+
+import static graphql.schema.GraphQLList.list;
+
+import graphql.schema.GraphQLInputObjectField;
+import graphql.schema.GraphQLInputObjectType;
+import graphql.schema.GraphQLInputType;
+import graphql.schema.GraphQLList;
+import graphql.schema.GraphQLNonNull;
+import graphql.schema.GraphQLScalarType;
+import io.stargate.db.schema.Column;
+
+/**
+ * Caches GraphQL field "filter input" types, for example 'StringFilterInput' in:
+ *
+ * <pre>
+ * input BooksFilterInput {
+ *   author: StringFilterInput
+ *   title: StringFilterInput
+ * }
+ * </pre>
+ *
+ * These are used when receiving filtering criteria, for example in a 'books' query:
+ *
+ * <pre>
+ * query {
+ *   books(filter: { author: { eq: "Cormac McCarthy" } }) { ... }
+ * }
+ * </pre>
+ *
+ * Note that these filters are also used in "DELETE IF" and "UPDATE IF" conditions. Some operators
+ * only apply in one context, for example "notEq" is allowed in conditional updates, but not
+ * selects. We currently have a single filter type with all the operators, it's up to the user to a
+ * valid one otherwise the query will fail. We could possibly refine this by creating multiple
+ * filter types depending on the context (whether the column is a collection, is a PK, has an SAI
+ * index defined for it, etc).
+ */
+public class FieldFilterInputTypeCache extends FieldTypeCache<GraphQLInputType> {
+
+  private final FieldInputTypeCache inputTypeCache;
+
+  FieldFilterInputTypeCache(FieldInputTypeCache inputTypeCache, NameMapping nameMapping) {
+    super(nameMapping);
+    this.inputTypeCache = inputTypeCache;
+  }
+
+  @Override
+  protected GraphQLInputType compute(Column.ColumnType columnType) {
+    GraphQLInputType gqlInputType = inputTypeCache.get(columnType);
+    if (columnType.isMap()) {
+      return mapFilter(gqlInputType);
+    } else if (columnType.isList() || columnType.isSet()) {
```

Comment:
Really a matter of taste, but FWIW I find that the `else` emphasize that we have 3 different branches. For example if I forgot the `return` in the second branch, the code would not compile, instead of failing silently.